### DISCUSSION TITLE
Problem with link understanding + pin setup.

### DISCUSSION THREAD

**Discussion by majer565** [Ardulink/Ardulink-2#103] [Q&A]
He devs, I couldn't find an answer for a long time so I decided to ask here. I want to control stepper motor with arduinoUNO and cnc shield with java so I found your repo and I am  curious how could I code that (if thats possible ofc). I see I need to use `Link ardLink = Links.getDefault()` but what should I do next? How can I retrive **boolean** information if there is a connection? Next, to control the steppers I need to set pins as outputs (there is a method `pinMode(pin, OUTPUT)` in arduino ) and `digitalWrite(pin, LOW/HIGH)` to write the pin. Here is my question how could I do that with Ardulink-2? Thanks for you time.

**Comment by pfichtner** [comment]
If you want to control the stepper motor without a library ([see examples like this](https://howtomechatronics.com/tutorials/arduino/stepper-motors-and-arduino-the-ultimate-guide/)), see the [micro example code how to turn on and off digital pins](https://github.com/Ardulink/Ardulink-2/). Beside the delay function ([sleep method](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html#sleep(long))) that should be everything you need. 

If you get a Connection (Link) there is a connection (otherwise an exception is thrown).

**Comment by majer565** [comment]
Yes just without library. So basically if I want to use ardulink-2 I do not need to set the output like with `pinMode(pin, OUTPUT)`, just can use `link.switchDigitalPin(pin, power)` or did I misunderstand?

**Comment by pfichtner** [comment]
That's right, as you can see [here](https://github.com/Ardulink/Ardulink-2/blob/c0347915ccfa1dace60317e6d98f3b80f4f4b5f9/deploy-dist/rootfolder/sketches/ArdulinkProtocol/ArdulinkProtocol.ino#L96) and [here](https://github.com/Ardulink/Ardulink-2/blob/c0347915ccfa1dace60317e6d98f3b80f4f4b5f9/deploy-dist/rootfolder/sketches/ArdulinkProtocol/ArdulinkProtocol.ino#L102), if you alter pin states via Ardulink, the sketch will set those pin as output pins automatically.

**Comment by pfichtner** [comment]
No further reaction for within two years
